### ISSUE TITLE
f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.001

### ISSUE THREAD

**Issue by cecilehannay** [NCAR/amwg_dev#100]
**Description:**
Same as #53 
+ Includes a fix to prevent division by zero identified in [ESCOMP/CAM issue #543](https://github.com/ESCOMP/CAM/issues/543). Also includes some clean up of the definition of PBL height and ZM2 mixing height.

**Case directory:**
Locally (if still available):
/glade/p/cesmdata/cseg/runs/cesm2_0/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.001

On github:
https://github.com/NCAR/amwg_dev/tree/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.001

**Sandbox:**
Locally (if still available):
/glade/work/hannay/cesm_tags/cam6_3_041

On github:
https://github.com/ESCOMP/CAM/tree/cam6_3_041
hash: 963ebdc

**Diagnostics:**
AMWG diags (if available)
[https://webext.cgd.ucar.edu/FWscHIST/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.001/atm/](https://webext.cgd.ucar.edu/FWscHIST/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.001/atm/)

**Contacts:**
@swrneale, @cecilehannay

**Comment by cecilehannay** [comment]
@swrneale: I ran the diags. The impact of the bugfix is not insignificant. https://webext.cgd.ucar.edu/FWscHIST/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.001/atm/
Is it what you expected?

**Comment by swrneale** [comment]
I would say there are some quite significant changes actually! I expected few changes over the ocean, but changes over the orographic tropics that should be smoothed out in the new run which is good I think. There are some pretty hefty decreases in precip. over E. Africa and Madagascar, but they are very beneficial changes. I need to check the PDF of parcel properties from the data to make sure the scheme is behaving the way it should. Thanks!

**Comment by cecilehannay** [comment]
Does it mean we should use this as the new baseline and start future runs from that?

**Comment by swrneale** [comment]
I think so. It has made a pretty large improvement in precip. the skill is much improved. Would need to check other things first though.

**Comment by adamrher** [comment]
> I expected few changes over the ocean, but changes over the orographic tropics that should be smoothed out in the new run which is good I think.

agreed. the prect diff plots do tend to align with topographic features ... which makes sense since the bug would have mostly impacted regions well above sea level.

**Comment by swrneale** [comment]
@cecilehannay  any chance you could re-run the first year (1981), but output the same set of diags. as f.e21.FWscHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf (specifically the h1 stream) as I need to check on some of the more detailed convection tendencies? Do I have to enter a run request?

**Comment by cecilehannay** [comment]
@swrneale: Do you need 1981? The first year is 1979. Would it work? Or is there a reason you need 1981?

**Comment by swrneale** [comment]
@cecilehannay Yep that's fine. Sorry, I was looking at the first year of the climos.

**Comment by swrneale** [comment]
@cecilehannay Also could you try building the run, at least initially, with debug on, to see that it does not produce divide by zero problems, and runs for a month or so.

**Comment by swrneale** [comment]
@cecile Where are the climo files being put?

**Comment by cecilehannay** [comment]
@swrneale:
 
I ran 1 year with the extra output last night.
See: https://github.com/NCAR/amwg_dev/issues/104
The output is in:
/glade/scratch/hannay/archive/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.002

I am also running 1 month in DEBUG mode. It will take a bit longer. The logs files will be in: 
/glade/scratch/hannay/archive/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.002_db 

Climos are in:
/glade/p/cgd/amp/people/hannay/amwg/climo2
As I am using the CESM postprocessing tool for those, they are in the format expected by the tool. The naming is a bit different from what we had with the AMWG diags:
/glade/p/cgd/amp/people/hannay/amwg/climo

**Comment by swrneale** [comment]
@cecilehannay awesome thanks! @islasimpson this simulation has rearranged convection quite a bit in the tropics and so I was wondering whether you could do your tape recorder plot if you get change? Thanks!

**Comment by islasimpson** [comment]
I've added it on at the end [here](https://www.cgd.ucar.edu/staff/islas/watervapor/taperecorder_jan2022.pdf)

**Comment by swrneale** [comment]
Thanks @islasimpson. Nothing too dramatically different then.

**Comment by justin-richling** [comment]
@swrneale, @cecilehannay

ADF diags: [https://webext.cgd.ucar.edu/FWscHIST/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.001/atm/ADF/](https://webext.cgd.ucar.edu/FWscHIST/f.c6_3_41.FWscHIST.ne30_L58.zm2_fix.001/atm/ADF/)
